### PR TITLE
chore: migrate project linting from 'tslint' to 'eslint'

### PR DESCRIPTION
**Pull request by MaxKless**
## PR Type

[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no API changes)
[x] Build-related changes
[x] CI-related changes
[ ] Documentation content changes
[ ] Application / infrastructure changes
[ ] Other: 

## What Is the Current Behavior?

Currently, we use the deprecated tslint. This is no langer possible with an update to Angular 13 that completely removes the tslint support.

## What Is the New Behavior?

Now, eslint is used.

## Does this PR Introduce a Breaking Change?

[x] Yes

## Other Information

[AB#66141](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/66141)

### CODE REVIEW

**Comment by cbratsch** [comment]
Rule fixer for sorting imports doesn't work, when first line is empty.
Please check.

File: eslint-rules/src/rules/use-jest-extended-matchers-in-tests.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import { AST_NODE_TYPES, TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+
+import { getClosestAncestorByKind } from '../helpers';
+
+const REPLACEMENTS = [
```
Comment (Eisie96) [inline eslint-rules/src/rules/use-jest-extended-matchers-in-tests.ts:5]: Why is this configuration directly included in the lint rule instead as a configuration in the eslintrc.json?

File: eslint-rules/src/rules/use-jest-extended-matchers-in-tests.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    if (!context.getFilename().endsWith('.spec.ts')) {
+      return {};
+    }
+    const [options] = context.options;
+    const mergedReplacements = [
+      ...REPLACEMENTS,
+      ...(options?.map(rep => ({ ...rep, pattern: new RegExp(rep.pattern) })) ?? []),
+    ];
+    return {
+      'Identifier[name=/(toBe|toEqual)/]'(node: TSESTree.Identifier) {
```
Comment (Eisie96) [inline eslint-rules/src/rules/use-jest-extended-matchers-in-tests.ts:62]: Should this rule always check toBe or toEqual calls? Could this be also a configuration?

**Review by Eisie96 — CHANGES_REQUESTED** [review]

File: eslint-rules/src/rules/no-assignment-to-inputs.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  meta: {
+    messages: {
+      inputAssignmentError: `Assigning to @Input decorated properties is forbidden.`,
+    },
+    type: 'problem',
+    schema: [],
+  },
+  create: context => {
+    // only apply to component files
+    if (context.getFilename().search(/.(component|container).ts/) < 0) {
```
[the remark this answers is not in the document]
Comment (MaxKless) [reply eslint-rules/src/rules/no-assignment-to-inputs.ts:15]: `isComponent` checks a `ClassDeclaration` node but this is before we have access to any nodes based purely on the file name.

File: eslint-rules/src/rules/use-alias-imports.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * @returns pairs of path patterns and alias strings, which can be resolved by the tsconfig.json
+ */
+function getAliasImports(): { pattern: string; alias: string }[] {
+  try {
+    const config = parse(fs.readFileSync('./tsconfig.json', { encoding: 'utf-8' }));
+    if (config && config.compilerOptions && config.compilerOptions.paths) {
+      const paths = config.compilerOptions.paths;
+      return Object.keys(paths).map(key => ({
+        pattern: `.*/${paths[key][0].replace(/\/\*$/, '/')}`,
```
[the remark this answers is not in the document]
Comment (MaxKless) [reply eslint-rules/src/rules/use-alias-imports.ts:77]: as talked about, skipping this one because it doesn't rly makes sense

**Review by Eisie96 — CHANGES_REQUESTED** [review]

**Review by Eisie96 — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -113,18 +120,41 @@
...
+    "eslint-plugin-ngrx": "^1.28.0",
+    "eslint-plugin-no-null": "1.0.2",
+    "eslint-plugin-prefer-arrow": "1.2.3",
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-react": "7.27.0",
+    "eslint-plugin-rxjs": "4.0.2",
+    "eslint-plugin-rxjs-angular": "^1.0.6",
+    "eslint-plugin-sonarjs": "^0.10.0",
+    "eslint-plugin-unicorn": "37.0.1",
+    "eslint-plugin-unused-imports": "^2.0.0",
```
[the remark this answers is not in the document]
Comment (shauke) [reply package.json]: The dependencies update will be done with the Angular 13 update. We will start with this set for now.

**Review by marschmidt89 — DISMISSED** [review]
documentation review done

**Review by shauke — APPROVED** [review]
Very nice migration. I would say this should be all right now and ready to be merged.

**Pull request merged**